amarna/rules/gatherer_rules/AllFunctionCallsGatherer.py

```diff
@@ -11,7 +11,9 @@ class FunctionCallType:
     """Represents a function call."""
 
     file_name: str
+    parent_function: str
     function_name: str
+    tail_name: str
```

Comment:
Added these additional fields to the function call gatherer in order to use it later in the post process rule.